frappe/contacts/doctype/address/address.py

```diff
@@ -237,6 +237,19 @@ def address_query(doctype, txt, searchfield, start, page_len, filters):
 			value=value
 		)
 
+	meta = frappe.get_meta("Address")
+	searchfields = meta.get_search_fields()
+
+	if searchfield:
+		searchfields.append(searchfield)
+
+	search_condition = ''
+	for field in searchfields:
```

Comment:
Code lines can be reduced by
```suggestion
	for field in searchfields:
		search_condition += '`tabAddress`.`{field}` like %(txt)s OR'
			.format(field=field)
	search_condition.rstrip(' OR')
```

Reply:
I think it would be simpler and more readable to store all the conditions in an array, and then just do a `join`.